frontends/election-manager/src/screens/print_test_deck_screen.tsx

```diff
@@ -259,237 +228,199 @@ function PrintingModal({
-      await printer.print({ sides: 'one-sided' });
+      await printElement(PrecinctTallyReport({ election, precinctId }), {
+        sides: 'one-sided',
+      });
       await logger.log(LogEventId.TestDeckTallyReportPrinted, userRole, {
         disposition: 'success',
         message: `Test deck tally report printed as part of L&A package for precinct ID: ${precinctId}`,
         precinctId,
       });
-
-      await makeCancelable(sleep(numPages * ONE_SIDED_PAGE_PRINT_TIME_MS));
-      advancePrinting();
+      await makeCancelable(sleep(numParties * ONE_SIDED_PAGE_PRINT_TIME_MS));
     },
-    [
-      advancePrinting,
-      printIndex,
-      printer,
-      logger,
-      userRole,
-      precinctIds,
-      makeCancelable,
-    ]
+    [election, logger, makeCancelable, userRole]
   );
 
-  const onAllBmdPaperBallotsRendered = useCallback(
-    async (numBallots) => {
-      if (!printIndex) {
-        return;
-      }
-
-      const precinctId = precinctIds[printIndex.precinctIndex];
-      await printer.print({ sides: 'one-sided' });
+  const printBmdPaperBallots = useCallback(
+    async (precinctId: PrecinctId) => {
+      const bmdPaperBallots = getBmdPaperBallots({
+        electionDefinition,
+        precinctId,
+      });
+      await printElement(<React.Fragment>{bmdPaperBallots}</React.Fragment>, {
+        sides: 'one-sided',
+      });
       await logger.log(LogEventId.TestDeckPrinted, userRole, {
         disposition: 'success',
         message: `BMD paper ballot test deck printed as part of L&A package for precinct ID: ${precinctId}`,
         precinctId,
       });
-
-      await makeCancelable(sleep(numBallots * ONE_SIDED_PAGE_PRINT_TIME_MS));
-      advancePrinting();
+      await makeCancelable(
+        sleep(bmdPaperBallots.length * ONE_SIDED_PAGE_PRINT_TIME_MS)
+      );
     },
-    [
-      advancePrinting,
-      userRole,
-      logger,
-      makeCancelable,
-      precinctIds,
-      printer,
-      printIndex,
-    ]
+    [electionDefinition, logger, makeCancelable, userRole]
   );
 
-  const onAllHandMarkedPaperBallotsRendered = useCallback(
-    async (numBallots) => {
-      if (!printIndex) {
-        return;
-      }
-
-      const precinctId = precinctIds[printIndex.precinctIndex];
-      await printer.print({ sides: 'two-sided-long-edge' });
+  const printHandMarkedPaperBallots = useCallback(
+    async (precinctId: PrecinctId, isLastPrecinct: boolean) => {
+      const handMarkedPaperBallotsWithCallback =
+        getHandMarkedPaperBallotsWithOnReadyCallback({
+          election,
+          electionHash,
+          precinctId,
+        });
+      const numBallots = handMarkedPaperBallotsWithCallback.length;
+
+      await printElementWhenReady(
+        (onAllRendered) => {
+          let numRendered = 0;
+          function onRendered() {
+            numRendered += 1;
+            if (numRendered === numBallots) {
+              onAllRendered();
+            }
+          }
```

Comment:
Remember the pattern we used in the print helpers of creating a promise for each subtask, passing the `resolve` function as the `onRendered` callback, and then using `Promise.all` to wait for them all? That could be nice here

Reply:
It seems a little less straightforward in this case (I'm thinking I might have to have a `useEffect` to wait for all the promises). Maybe it's simpler than I'm thinking, but since this works and is intelligible, I'll leave it as is.